interfaces/builtin/optical_drive.go

```diff
@@ -28,27 +37,73 @@ const opticalDriveBaseDeclarationSlots = `
         - core
 `
 
-const opticalDriveConnectedPlugAppArmor = `
-# Allow read access to optical drives
-/dev/sr[0-9]* r,
-/dev/scd[0-9]* r,
-@{PROC}/sys/dev/cdrom/info r,
-/run/udev/data/b11:[0-9]* r,
-`
```

Comment:
I prefer that most of the rules be in a single variable and then conditionally add the write rules, which is more clear. Eg:

```
const opticalDriveConnectedPlugAppArmor = `
# Allow read access to optical drives
/dev/sr[0-9]* r,
/dev/scd[0-9]* r,
@{PROC}/sys/dev/cdrom/info r,
/run/udev/data/b11:[0-9]* r,
`
...
func (iface *opticalDriveInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
       var write bool
       _ = plug.Attr("write", &write)

       // Add the common readonly policy
       spec.AddSnippet(opticalDriveConnectedPlugAppArmor)

       // 'write: true' grants write access to the devices
       if write {
               spec.AddSnippet("# Allow write access to optical drives")
               spec.AddSnippet("/dev/sr[0-9]* w,")
               spec.AddSnippet("/dev/scd[0-9]* w,")
       }
       return nil
}
```